src/dot.ts

```diff
@@ -133,17 +133,13 @@ export class Dot extends Modifier {
 
     this.position = Modifier.Position.RIGHT;
 
-    this.radius = 2;
-    this.setWidth(5);
+    this.setText(Glyphs.augmentationDot);
     this.dotShiftY = 0;
   }
 
   setNote(note: Note): this {
     this.note = note;
-    if (isGraceNote(note)) {
-      this.radius *= 0.5;
-      this.setWidth(3);
-    }
+    this.font = note.font;
```

Comment:
So how does it look now for grace notes with a dot?

Is the dot full sized? The original code makes the dot half sized for grace notes.

Reply:
The note, if it is a grace note, should already be scaled. The recommended scale is not 0,5 but 3/5 or 2/3 I do not remember now.

Reply:
By the way, what would be the meaning of a dot on a grace note? Do you know if this possible? In any case if a dot is to be presented in a note, I think that it makes sense that they use the same font.

Reply:
Yeah I have no idea what it would mean to add a dot here. But maybe Grace Note in this case just refers to notes that are rendered in a smaller font (e.g., a voice that is rendered smaller than the main voice)? In any case, I pointed it out just because it was deleted here. I figure it was put in for a good reason at some point. ;-)

Reply:
I found this jsfiddle from the tutorials: https://jsfiddle.net/ronyeh/smyht3q5/

So apparently you can just stick GraceNotes anywhere, to indicate maybe a separate (less important) voice? What if we wanted to add a dot to those notes? Will the dot be magically scaled down?

Reply:
@mscuthbert @AaronDavidNewman do dots make sense on grace notes?
@ronyeh I think that they will scale down, I wil add a test if the combination make sense

Reply:
Though rare, I've definitely seen grace notes with dots.  Particularly in bagpipe music where you might have a dotted-eighth + 16th + eighth grace figure to show that the figure should be played as quickly as possible before the main note.  The W3C music notation group has been talking about graces a lot recently RE: MNX and basically almost anything you've seen in a score can happen in a grace group (we were talking most recently about the thorny problem of clef changes within a grace group, especially if there are two voices in a measure with different numbers of grace notes at the same time where a clef change happens mid grace group).

Morton Feldman--I believe in _The Viola in My Life_--put a fermata on a grace note.  Once you do that, you can do anything.

Reply:
@mscuthbert could you provide a picture. I tried to find it but I did not succeed

Reply:
https://forums.steinberg.net/t/small-problem-with-embellishments/842546

Reply:
And they are frequent in cadenzas which take no metrical time and thus have the same positioning and alignment problems as grace note groups.  (and they can themselves have grace notes on them, such as at the end of the most commonly played cadenza to the Mozart clarinet concerto).

Reply:
Thanks a lot. And could you provide a picture of the fermata case?